### DISCUSSION TITLE
using runGraphInFile in .js script, using "Loops" Tutorial Graph, but no user input window.

### DISCUSSION THREAD

**Discussion by hadupa** [Ironclad/rivet#434] [General]
Just like the title says, I'm following the user guide and trying to use the runGraphInFile function to test out graph execution from a script call before integrating Rivet into my larger project. I execute the script (shown below). test_project is a rivet project I created wherein I copy/pasted the "Loops" graph found in the tutorial project provided on the Rivet site. I am using the following short code in my test script:

import { runGraphInFile } from '@ironclad/rivet-node';

const outputs = await runGraphInFile('./test_project.rivet-project', {
  graph: 'Loops',
  // other options...
});

I removed the other options for the runGraphInFile function call, so it wouldn't surprise me if I need to configure that (wanted to minimize potential issues for my test). 

I execute the script using my CLI:
node my_script.js

After a few minutes the program exits with no sign that there was an error. However, the user input window never appears and I am never prompted to input text (see "Loops" example in Tutorial). The script simply executes then closes.

**Comment by abrenneke** [comment]
Yeah, user input has to be handled manually when running Rivet in a parent script. There's no connection for it with the remote debugger. You can implement user input manually (say in a console app) by handling the `onUserInput` event in your code, then calling `context.userInput` as your callback, but it's a bit awkward to do so. External functions are the preferred way to handle things like that when you have a parent script.

**Reply by hadupa** [reply]
Thanks for the fast response. I was afraid of that, but should be easy enough to resolve. Great program!
